view/frontend/templates/button_product_page.phtml

```diff
@@ -207,31 +202,35 @@ if (!$block->isSupportableType()) return;
                         options: JSON.stringify(options)
                     }]
                 };
+
+                var hintsPromise = new Promise(function (resolve, reject) {
```

Comment:
Maybe we have a little problem. If we hints ajax call works long enough (10 seconds), user can click product options, configureProductCheckout calls again, hints call again.
Bolt modal wait for last ajax call and can as result show timeout error.
Maybe it's better to create promise only once, not on each setupProductPage() call

Reply:
@vitaliyreznikov Can you create a backlog task in the M2 asana board to investigate the identified corner case?

Reply:
@matt-thomason Sure. https://app.asana.com/0/941920570700290/1165280808501785/f